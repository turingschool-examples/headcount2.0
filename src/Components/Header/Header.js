import React from 'react'
import icon from '../../assets/bee.svg'
import './header.css'
// import Search from '../Search/Search'

const Header = () => {
    return (
        <div className="header">
            <div className="logo">
                <img  className="logo-img" alt="logo" src={icon}></img>
                <h1 className="logo-text">Head Count 2.0</h1>
            </div>
        </div>
    )
}

export default Header