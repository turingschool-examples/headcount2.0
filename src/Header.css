/*HEADER COMPONENT RELATED CSS GOES HERE*/
