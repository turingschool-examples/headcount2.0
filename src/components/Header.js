import React from 'react';
import PropTypes from 'prop-types';

import InputField from './InputField';

const Header = (props) => {
  return (
    <header>
      <img className='logo' src="./brain-and-head.svg" alt="logo" />
      <h1 className='banner-title'>Colorado Kindercount</h1>
      <img 
        className='info-btn'
        src='./information.svg'
        alt='info-button'
        onClick={() => {
          props.toggleModal();
        }}
      />
      <InputField 
        processFilter={props.processFilter}
      />
    </header>
  );
};

Header.propTypes = {
  processFilter: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired
};

export default Header;