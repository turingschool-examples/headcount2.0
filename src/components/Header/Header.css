/*HEADER COMPONENT RELATED CSS GOES HERE*/
.Header {
  padding: 1rem;
  grid-row: 1 / 2;
  grid-column: 1 / 2;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  background-color: #DCF1F4;
  color: #006989;
  text-align: center;
  font-family: 'Roboto', sans-serif;
  font-weight: 900;
}

h1 {
  margin: 50px 0 20px 0;
  padding: 0;
  grid-row: 1 / 2;
  font-size: 3rem;
}

.count {
  color: #009FB7;
}

.version {
  color: #9E9E9E;
  font-family: 'Roboto Condensed', sans-serif;
  font-weight: 400;
}


.Header h2 {
  margin: 40px 0 0 0;
  font-family: 'Roboto', sans-serif;
  font-weight: 900;
}

p {
  margin-bottom: 40px;
  font-weight: 500;
  font-size: 1.15rem;
}