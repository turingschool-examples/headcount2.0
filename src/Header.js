import React from 'react'
import Search from './Search'
import './Header.css'

const Header = () => {
  return(
    <header className="Header">
      <h1>HeadCount 2.0</h1>
      <Search />
    </header>
  )
}

export default Header