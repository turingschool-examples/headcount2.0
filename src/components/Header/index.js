import React from 'react'
import PropTypes from 'prop-types'
import Search from '../Search'
import './Header.css'

const Header = ({findAllMatches}) => {
  return(
    <header className="Header">
      <h1>Head
        <span className="count">Count</span>
        <span className="version"> 2.0</span>
      </h1>
      <Search findAllMatches={findAllMatches} />
      <h2>Click on two districts below to compare averages.</h2>
      <p>If you know the name of the districts you would like to compare,<br /> type them in the search field above to locate them more quickly.</p>
    </header>
  )
}

Header.propTypes = {
  findAllMatches: PropTypes.func.isRequired
}

export default Header