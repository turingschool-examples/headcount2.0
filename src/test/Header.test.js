import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Header from '../components/Header';


describe('Header', ()=>{
  let wrapper; 

  const mockToggleModal = jest.fn();
  const mockProcessFilter = () => {

  }

  beforeEach(()=>{
    wrapper = shallow(<Header 
      toggleModal={mockToggleModal}
      processFilter={mockProcessFilter}
    />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render content to the DOM', () => {
    expect(wrapper).toMatchSnapshot();    
  });

  it('should call toggleModal on button click', () => {
    wrapper.find('.info-btn').simulate('click')

    expect(mockToggleModal.mock.calls.length).toBe(1)
  });

});