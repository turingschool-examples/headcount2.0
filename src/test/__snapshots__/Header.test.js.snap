// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render content to the DOM 1`] = `
<header>
  <img
    alt="logo"
    className="logo"
    src="./brain-and-head.svg"
  />
  <h1
    className="banner-title"
  >
    Colorado Kindercount
  </h1>
  <img
    alt="info-button"
    className="info-btn"
    onClick={[Function]}
    src="./information.svg"
  />
  <InputField
    processFilter={[Function]}
  />
</header>
`;
